crates/daphne-server/src/storage_proxy_connection/kv/mod.rs

```diff
@@ -256,20 +256,54 @@ impl<'h> Kv<'h> {
         skip_all,
         fields(key, prefix = std::any::type_name::<P>()),
     )]
-    pub async fn put<P>(&self, key: &P::Key, value: P::Value) -> Result<(), Error>
-    where
-        P: KvPrefix,
-        P::Key: std::fmt::Debug,
+    pub async fn put<P>(
+        &self,
+        key: &P::Key,
+        value: P::Value,
+    ) -> Result<(), Error>
+        where
+            P: KvPrefix,
+            P::Key: std::fmt::Debug,
+    {
+        self.put_with_expiration::<P>(key, value, None).await
+    }
+
+    // New method with expiration parameter
+    #[tracing::instrument(
+        name = "kv_put",
+        skip_all,
+        fields(key, prefix = std::any::type_name::<P>()),
+    )]
+    pub async fn put_with_expiration<P>(
+        &self,
+        key: &P::Key,
+        value: P::Value,
+        expiration: Option<u64>,
+    ) -> Result<(), Error>
+        where
+            P: KvPrefix,
+            P::Key: std::fmt::Debug,
     {
         let key = Self::to_key::<P>(key);
         tracing::debug!(key, "PUT");
+
+        // Create the body with optional expiration
+        let mut body = serde_json::json!({
+            "value": value,
+        });
+
+        if let Some(exp) = expiration {
+            body["expiration"] = serde_json::json!(exp);
+        }
+
```

Comment:
This has to be done in the storage_proxy, I think to keep things backwards compatible I'd pass the expiration through a header like `X-expiration` or something close to that